tools/codegen/api/python.py

```diff
@@ -173,6 +173,53 @@
 #      }
 #
 
+# TODO: stick this more firmly in the data model somewhere?
+def namedtuple_fieldnames(returns: Tuple[Return]) -> List[str]:
+    if len(returns) <= 1 or all(map(lambda r: r.name is None, returns)):
+        return []
+    else:
+        if any(map(lambda r: r.name is None, returns)):
+            # When building on Windows, `PyStructSequence_UnnamedField` could not be
+            # resolved by the linker for some reason, which cause error in building:
+            #
+            # python_nn_functions.cpp.obj : error LNK2001: unresolved external symbol
+            # PyStructSequence_UnnamedField
+            #
+            # Thus, at this point in time, we do not support unnamed
+            # fields in namedtuple; you must either name all fields,
+            # or none of them.
+            raise ValueError("Unnamed field is not supported by codegen")
+
+        return list(map(lambda r: str(r.name), returns))
+
+@dataclass(frozen=True)
+class PythonReturns:
+    returns: Tuple[Return]
+
+    def named_tuple(self) -> Tuple[str]:
+        python_returns = [argument_type_str_pyi(r.type) for r in self.returns]
+        field_names = namedtuple_fieldnames(self.returns)
+        if field_names:
+            namedtuple_name = '_'.join(['namedtuple'] + field_names)
+            tuple_args = ['("{}", {})'.format(name, typ) for name, typ in zip(field_names, python_returns)]
+            namedtuple_def = 'NamedTuple("{}", [{}])'.format(namedtuple_name, ', '.join(tuple_args))
+            return namedtuple_name, namedtuple_def
+        return ()
+
+    def returns_str(self) -> str:
```

Comment:
Best to suffix this with `_pyi` to make it clear that it is generating pyi type stub format

Reply:
same with above, I think